### ISSUE TITLE
PROPOSAL: Presentaton mode

### ISSUE THREAD

**Issue by rgardler** [rgardler/simdem#19]
Using reveal.js we could create a browser based presentation from the markdown document. This would allow the content to be presented in a browser alongside a command shell running in tutorial or demo mode.

# Proposed markdown syntax

We will need to separate slide content from notes, as well as separate parent and child slides:

Parent slides separator: '-- In Summary --'
Child slide separator: '-- Digging In --'
Slide detail/notes separator: '-- In Detail --'

In tutorial mode these separators could be handled as follows:

'-- In Summary --' Shown by default
'-- Digging In --' Ignored, the tutorial will flow through the "digging in" sections
'-- In Detail --' Will be displayed as a '[More Detail]' tag, the user can press 'm' to see the contents of this section

**Comment by rgardler** [comment]
The problem with this proposal is that the separators will be visible in any document generated from the markdown. A better approach would be to use [Markdown comments](https://stackoverflow.com/questions/4823468/comments-in-markdown). These will no appear in the final rendered HTML.

**Comment by suhaildawood** [comment]
It would be great to expand this feature and have a side by side presentation component and Azure Cloud Console components, where you can login to Azure and run the commands without needing a shell with the Azure CLI installed and configured.

**Comment by suhaildawood** [comment]
Alternatively, we could also opt for a command line implementation such as [mdp](https://github.com/visit1985/mdp)

**Comment by rgardler** [comment]
I agree, but I think it would be best to keep the two features separate from an implementation perspective. They can, and should, be implemented separately so as not to have one block the other. That is, it would be good to create a separate issue for the creation of a browser based console application.

**Comment by rgardler** [comment]
I've been thinking about this and I just don't think it is going to work. A tutorial is not a presentation. Perhaps a better approach is actually to merge the two ideas as @suhaildawood  suggests. We already have a logical structure in documents using headings to separate sections. If we have a browser view that has both a console to execute commands and the descriptive text that matches the command we can create a good presentation view.

**Comment by rgardler** [comment]
Perhaps the best candidate for a console view is XTerm.js, see https://xtermjs.org/
